### PR TITLE
Feature/dom hydro

### PR DESCRIPTION
**Pull request by st0nebreaker**
#### What's this PR do?
Hydration DOM - findOuncesForDay() method runs

#### Where should the reviewer start?
local machine

#### How should this be manually tested?
open src/index.html to check DOM display

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -8,51 +9,45 @@ let avgH2o = document.querySelector('.h2o-avg');
...
 
-let userArray = [];
-
-
-userData.forEach(user => {
-	let currentUser = new User(user);
-	userArray.push(currentUser);
-})
-
+const userArray = userData.map((user) => new User(user));
```
Comment (st0nebreaker) [inline src/scripts.js:12]: shorter to use .map() than .forEach() pushing into an empty array

File: src/scripts.js [hunk 2]
```diff
@@ -8,51 +9,45 @@ let avgH2o = document.querySelector('.h2o-avg');
...
+let currentHydration = {};
 
+const chooseRandom = () => {
+	const randomNum = (Math.floor(Math.random() * userRepo.data.length));
 
-let chooseRandom = () => {
-	let randomNum = (Math.floor(Math.random() * currentUserData.data.length));
-	randomUser = currentUserData.data[randomNum];
+	randomUser = userRepo.data[randomNum];
+	currentHydration = new Hydration(randomUser, hydrationData);
```
Comment (st0nebreaker) [inline src/scripts.js:21]: hydration class instance gets updated along with the randomUser, ensuring it's argument of randomUser is accessed properly

File: src/scripts.js [hunk 3]
```diff
@@ -8,51 +9,45 @@ let avgH2o = document.querySelector('.h2o-avg');
...
+	currentHydration.correctHydroData();
```
Comment (st0nebreaker) [inline src/scripts.js:22]: filters the list of 5,000 data into 100 (only user.id data)

File: src/scripts.js [hunk 4]
```diff
@@ -8,51 +9,45 @@ let avgH2o = document.querySelector('.h2o-avg');
...
+const updateWelcome = (currentUser) => {
+	welcomeTitle.innerText = `Welcome ${currentUser.name}!`;
+	userCardName.innerText = `Name: ${currentUser.name}`;
+	userAddress.innerText = `Address: ${currentUser.address}`;
+	userEmail.innerText = `Email: ${currentUser.email}`;
 }
 
-let updateHydration = () => {
-	let ouncesForDay = currentHydration.findOuncesForDay('2019/09/22');
+const updateHydration = (currentHydro) => {
```
Comment (st0nebreaker) [inline src/scripts.js:32]: passing in argument from line 49, to be more dynamic & match the fn setup on line 25

File: src/scripts.js (hunk 2 above)
Comment (st0nebreaker) [reply src/scripts.js:21]: this was the main problem from the last code, the randomUser was empty floating off in the ether when currentHydration was being made, so randomUser was empty. putting it in this function populates that value.

File: src/Hydration.js [hunk 5]
```diff
@@ -20,6 +20,7 @@ class Hydration {
 	}
 
 	findOuncesForDay(date) {
+		//user hydrodata empty
```
Comment (st0nebreaker) [inline src/Hydration.js:23]: oops left a comment in

File: src/scripts.js (hunk 2 above)
Comment (carlymclift) [reply src/scripts.js:21]: yay! I'm glad you got it figured out.

File: src/Hydration.js (hunk 5 above)
Comment (carlymclift) [reply src/Hydration.js:23]: I can remove it.

**Pull request merged**